src/error.rs

```diff
@@ -12,6 +12,8 @@ pub enum Error {
     DecompressingError(std::io::Error),
     /// An error occured when decoding a base64 encoded dataset.
     Base64DecodingError(base64::DecodeError),
+    /// An error occurred when decoding a csv encoded dataset.
+    CsvDecodingError(String),
```

Comment:
I'd prefer that this were an error type so the user could handle it (or format it some other way) if necessary

Reply:
I think we have several options here. One option would be

```rs
CsvDecodingError(Box<dyn std::error::Error + Send + Sync + 'static>)
```

Alternatively, we could do something like

```rs
/// Errors that can occur while decoding csv data.
#[derive(Clone, Debug, PartialEq, Eq)]
pub enum CsvError {
    TileDataParseError(ParseIntError)
}

/// Errors which occurred when parsing the file
#[derive(Debug)]
#[non_exhaustive]
pub enum Error {
<SNIP>
    /// An error occurred when decoding a csv encoded dataset.
    CsvDecodingError(CsvError),
```

Do you have a preference?

Reply:
Second one looks better IMO: Lets you look into the error, like in the other enum variants.